bioptim/dynamics/configure_problem.py

```diff
@@ -312,14 +348,75 @@ def configure_contact_function(ocp, nlp, dyn_func: Callable, **extra_params):
             legend=all_contact_names,
         )
 
+    @staticmethod
+    def configure_soft_contact_function(ocp, nlp):
+        """
+        Configure the soft contact sphere
+
+        Parameters
+        ----------
+        ocp: OptimalControlProgram
+            A reference to the ocp
+        nlp: NonLinearProgram
+            A reference to the phase
+        """
+
+        global_soft_contact_force_func = MX.zeros(nlp.model.nbSoftContacts() * 6, 1)
+
+        for i_sc in range(nlp.model.nbSoftContacts()):
+            soft_contact = nlp.model.softContact(i_sc)
+            n = int(nlp.states.mx_reduced.shape[0] / 2)
+
+            global_soft_contact_force_func[i_sc * 6: (i_sc + 1) * 6, :] = (
+                biorbd.SoftContactSphere(soft_contact)
+                    .computeForceAtOrigin(nlp.model, nlp.states.mx_reduced[:n], nlp.states.mx_reduced[n:])
+                    .to_mx()
+            )
+        nlp.soft_contact_forces_func = Function(
+            "soft_contact_forces_func",
+            [nlp.states.mx_reduced, nlp.controls.mx_reduced, nlp.parameters.mx],
+            [global_soft_contact_force_func],
+            ["x", "u", "p"],
+            ["soft_contact_forces"],
+        ).expand()
+
+        for i_sc in range(nlp.model.nbSoftContacts()):
+            all_soft_contact_names = []
+            l = ["Mx", "My", "Mz", "Fx", "Fy", "Fz"]  # TODO: find a better place to hold this or define it in biorbd ?
```

Comment:
Please remove this #todo as it is the right place, apart from the fact that it can be outside the for loop